teamsyncd/teamsync.cpp

```diff
@@ -43,6 +59,32 @@ TeamSync::TeamSync(DBConnector *db, DBConnector *stateDb, Select *select) :
     }
 }
 
+/*
+ * Get interface name based on interface index
+ * Return ifindex in string format if interface don't exist, or no cache
+ *        else return the corresponding interface name.
+ */
+string TeamSync::ifindexToName(int ifindex)
+{
+    char ifName[TeamPortSync::MAX_IFNAME + 1] = {0};
+
+    /* nl_socket, cache create failed earlier */
+    if (!m_nl_sock) {
+        /* Returns ifindex as string / */
+        return to_string(ifindex);
+    }
+ 
+    /* Refill cache and check for ifindex */
+    nl_cache_refill(m_nl_sock ,m_link_cache);
+    if (rtnl_link_i2name(m_link_cache, ifindex, ifName, TeamPortSync::MAX_IFNAME) == NULL)
+    {
+        /* Returns ifindex as string / */
+        return to_string(ifindex);
```

Comment:
the same as before

Reply:
The idea was to give back the ifindex itself as the interface name and hence the caller function match will fail and ignore the NETLINK update, instead of proceeding with the TeamSync::addLag() API.

Reply:
I changed the API format so that now it checks the interface name in the API checkIfindexToName() and returns true/false depending on the interface name is present in kernel cache or not.